### DISCUSSION TITLE
Want to call options of p-dropdown dynamically method keeps on executing

### DISCUSSION THREAD

**Discussion by VivekParmar-18** [primefaces/community#1924] [PrimeNG]
------code

@if(report!==undefined && report!==null){
    <div class="font-medium text-3xl text-900 text-center font-bold">{{lrn.haA_LONG_NAME}}</div>   
    
        <div class="p-fluid p-formgrid grid mt-2 mb-2" *ngFor="let rep of report" >
            <div class="col-12 md:col-3 mt-3 ml-7 p-0">
                <label htmlFor="name3">{{rep.healtH_ASSESSMENT_QUESTIONS?.hsQ_QUESTION_TEXT}}</label>
            </div>
            <div class="col-12 md:col-5 p-0">
                @if(rep.healtH_ASSESSMENT_QUESTIONS.hsQ_DISPLAY_TYPE=='txts') {
                <input pInputText id="name3" type="text" [(ngModel)]="rep.ptR_RESPONSE_VALUE" />}
                @if(rep.healtH_ASSESSMENT_QUESTIONS.hsQ_DISPLAY_TYPE=='txtl') {
                <textarea pInputTextarea id="name3" type="text" [(ngModel)]="rep.ptR_RESPONSE_VALUE" ></textarea>}
                @if(rep.healtH_ASSESSMENT_QUESTIONS.hsQ_DISPLAY_TYPE=='ddl') {
                    <p-dropdown [options]="getDropdownOptions(rep.healtH_ASSESSMENT_QUESTIONS.hsQ_HQR_ID)"></p-dropdown>
                }
            </div>
            @if(rep.healtH_ASSESSMENT_QUESTIONS.hsQ_QC_FLAG_DONE=='Yes') {
                <div class="col-12 md:col-1 md:col-offset-1 p-0">
                <p-dropdown [options]="options" optionLabel="label" optionValue="value" [style]="{'width':'100%'}"></p-dropdown>
            </div>    }
    </div>
    }
    -----ts
  ViewReport(number1:any,number2:any){
  this.http.get<any>(baseurl+"Patient/GetReport?PTS_ID="+number1+"&PTR_HAA_ID="+number2).subscribe(res=>{
    this.report=res;
    console.log(this.report);
    res.forEach((element:any) => {
    if(element.healtH_ASSESSMENT_QUESTIONS.hsQ_HQR_ID!==null){
        this.http.get<any[]>(baseurl + 'Patient/GetResponseCode?HQR_RESPONSE_CODE=' + element.healtH_ASSESSMENT_QUESTIONS.hsQ_HQR_ID).subscribe(res => {
          this.dropdownOptionsMap[element.healtH_ASSESSMENT_QUESTIONS.hsQ_HQR_ID] = res.map((item: any) => ({ label: item.text, value: item.value }));
        });
      }});
    })
  }
 getDropdownOptions(hsQ_HQR_ID: string): any[] | undefined {
  console.log( this.dropdownOptionsMap[hsQ_HQR_ID])
  return this.dropdownOptionsMap[hsQ_HQR_ID];
}

**Comment by mehmetcetin01140** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue clearly?
